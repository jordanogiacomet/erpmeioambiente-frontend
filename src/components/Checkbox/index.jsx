import { CheckboxFieldComponent } from './CheckboxFieldComponent';
import { CheckboxLabelComponent } from './CheckboxLabelComponent';
import { CheckboxRoot } from './CheckboxRoot';

export const Checkbox = {
  Root: CheckboxRoot,
  Field: CheckboxFieldComponent,
  Label: CheckboxLabelComponent,
};