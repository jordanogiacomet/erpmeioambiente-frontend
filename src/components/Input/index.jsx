import { InputRoot } from './InputRoot';
import { InputLabel } from './InputLabel';
import { InputFieldComponent } from './InputFieldComponent';
import { InputIcon } from './InputIcon';

export const Input = {
  Root: InputRoot,
  Label: InputLabel,
  Field: InputFieldComponent,
  Icon: InputIcon,
};
