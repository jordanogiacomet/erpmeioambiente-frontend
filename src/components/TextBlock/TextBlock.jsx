import P from 'prop-types';
import styled from 'styled-components';
import Heading from '../Heading/Heading';

const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: clamp(300px, 50%, 600px);
  background-color: ${({ backgroundColor, theme }) => backgroundColor || theme.colors.primary};
  color: ${({ theme }) => theme.colors.secondary}
`;

const Paragraph = styled.p`
  margin-top: 16px;
  line-height: 2rem;
  font-size: ${({ theme }) => theme.font.sizes.large};
`;

const Subtitle = styled(Heading)`
  margin-top: 8px;
`;


const TextBlock = ({ children, title, subtitle, backgroundColor }) => {
  return (
    <TextContainer>
      <Heading>{title}</Heading>
      <Subtitle as="h3">{subtitle}</Subtitle>
      <Paragraph>{children}</Paragraph>
    </TextContainer>
  );
};

TextBlock.propTypes = {
  children: P.node.isRequired,
  title: P.string.isRequired,
  subtitle: P.string.isRequired,
  backgroundColor: P.string,
};

export default TextBlock;