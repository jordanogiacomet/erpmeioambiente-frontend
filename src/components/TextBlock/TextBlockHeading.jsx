import React from 'react';
import P from 'prop-types';
import { Heading } from '../Heading/Heading';

export const TextBlockHeading = ({ children, ...props }) => {
  return (
    <Heading {...props}>
      {children}
    </Heading>
  );
};

TextBlockHeading.propTypes = {
  children: P.node.isRequired,
};
