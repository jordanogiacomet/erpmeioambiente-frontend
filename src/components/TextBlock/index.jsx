import P from 'prop-types';
import styled from 'styled-components';
import Heading from '../Heading/Heading';

const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: clamp(300px, 50%, 600px);
  color: ${({ theme }) => theme.colors.secondary}
`;

const Paragraph = styled.p`
  margin-top: 16px;
  line-height: 2rem;
  font-size: ${({ theme }) => theme.font.sizes.large};
`;

const Subtitle = styled(Heading)`
  margin-top: 8px;
`;


export const TextBlock = ({ children, title, subtitle }) => {
  return (
    <TextContainer>
      <Heading>{title}</Heading>
      <Subtitle as="h3">{subtitle}</Subtitle>
      <Paragraph>{children}</Paragraph>
    </TextContainer>
  );
};

TextBlock.propTypes = {
  children: P.node.isRequired,
  title: P.string.isRequired,
  subtitle: P.string.isRequired,
};