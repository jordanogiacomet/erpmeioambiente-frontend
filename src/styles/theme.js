// theme.js
export const theme = {
  colors: {
    primary: '#000000',    // Manter o azul para destaques e bot√µes
    secondary: '#FFFFFF',  // Manter o branco para fundos ou textos claros
    background: '#E0F2F1', // Verde claro para o fundo principal
    text: '#004D40',       // Verde escuro para o texto
    border: '#80CBC4',     // Verde para bordas sutis
    focus: '#004D40',      // Cor de foco, pode ser usada para bordas ao focar no input
  },
  font: {
    sizes: {
      small: '1rem',
      medium: '1.2rem',
      large: '1.6rem',
      xlarge: '2rem',
      huge: '2.4rem',
      xhuge: '3rem',
    },
  },
  media: {
    min: {
      sm: '(min-width: 640px)',
      md: '(min-width: 768px)',
      lg: '(min-width: 1024px)',
      xl: '(min-width: 1280px)',
      xxl: '(min-width: 1536px)',
    },
    max: {
      sm: '(max-width: 640px)',
      md: '(max-width: 768px)',
      lg: '(max-width: 1024px)',
      xl: '(max-width: 1280px)',
      xxl: '(max-width: 1536px)',
    }
  },
};
