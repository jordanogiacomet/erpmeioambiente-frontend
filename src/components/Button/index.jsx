import { ButtonComponent } from './ButtonComponent';
import { ButtonWrapper } from './styles/ButtonWrapper';

export const Button = {
  Root: ButtonWrapper,
  Main: ButtonComponent,
};