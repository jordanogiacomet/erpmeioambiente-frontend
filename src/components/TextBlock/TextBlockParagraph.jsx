import React from 'react';
import P from 'prop-types';
import { Paragraph } from './styles/Paragraph';

export const TextBlockParagraph = ({ children, ...props }) => {
  return (
    <Paragraph {...props}>
      {children}
    </Paragraph>
  );
};

TextBlockParagraph.propTypes = {
  children: P.node.isRequired,
};
