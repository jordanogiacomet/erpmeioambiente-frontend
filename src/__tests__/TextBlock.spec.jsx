import React from 'react';
import { screen } from '@testing-library/react';
import renderTheme from '../styles/render-theme';
import { TextBlock } from '../components/TextBlock';
import theme from '../styles/theme';

describe('<TextBlock />', () => {
  it('should render all elements', () => {
    renderTheme(<TextBlock title="Title" subtitle="Subtitle">test</TextBlock>);

    expect(screen.getByRole('heading', { name: 'Title' })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: 'Subtitle' })).toBeInTheDocument();
    expect(screen.getByText('test')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<TextBlock title="Title" subtitle="Subtitle">Test</TextBlock>);

    expect(container).toMatchSnapshot();
  });
});