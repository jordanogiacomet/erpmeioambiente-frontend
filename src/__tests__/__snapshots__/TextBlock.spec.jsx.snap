// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBlock /> should match snapshot 1`] = `
.c1 {
  color: #FFFFFF;
  margin-bottom: 15px;
  font-size: 3rem;
  text-transform: none;
}

.c0 {
  display: flex;
  flex-direction: column;
  width: clamp(320px, 50%, 600px);
  color: #FFFFFF;
}

.c3 {
  margin-top: 16px;
  line-height: 2rem;
  font-size: 1.6rem;
}

.c2 {
  margin-top: 8px;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Title
    </h1>
    <h3
      class="c2"
    >
      Subtitle
    </h3>
    <p
      class="c3"
    >
      Test
    </p>
  </div>
</div>
`;
